en/docs/develop/restapis/oauth2-scope-endpoint.yaml

```diff
@@ -0,0 +1,375 @@
+host: localhost
+
+# The base path of the API.
+# Will be prefixed to all paths.
+
+# For a tenant, basepath should be /t/{tenant_domain}/api/identity/oauth2/v1.0.
+basePath: /api/identity/oauth2/v1.0
+#basePath: /t/{tenant_domain}/api/identity/oauth2/v1.0
+
+# The following media types can be passed as input in message bodies of the API.
+# The actual media type must be specified in the Content-Type header field of the request.
+# The default is json, i.e. the Content-Type header is not needed to
+# be set, but supporting it serves extensibility.
+consumes:
+  - application/json
+
+# The following media types may be passed as output in message bodies of the API.
+# The media type(s) consumable by the requestor is specified in the Accept header field
+# of the corresponding request.
+# The actual media type returned will be specfied in the Content-Type header field
+# of the of the response.
+# The default of the Accept header is json, i.e. there is not needed to
+# set the value, but supporting it serves extensibility.
+produces:
+  - application/json
+paths:
+  /scopes:
+    # Endpoint used to create scope
+    post:
+      description: |
+        This API is used to create a scope.
+      x-wso2-request: |
+        curl -X POST -H "Authorization: Basic YWRtaW46YWRtaW4=" -H "Content-Type: application/json" -d '{"name": "profile", "displayName": "profile", "description": "get all profile information", "bindings": ["role1", "role2"]}' "https://localhost:9443/api/identity/oauth2/v1.0/scopes"
+
+      x-wso2-response: |
+        "HTTP/1.1 201 Created"
+        {"name":"profile","description":"get all profile information","displayName":"profile","bindings":["role1","role2"]}
+
+      summary: |
+        Registers a Scope
+
+      operationId: registerScope
+
+      parameters:
+        - name: scope
+          in: body
+          description: Define a scope with bindings to register it
+          required: true
+          schema:
+            $ref: '#/definitions/scope'
+
+      responses:
+        201:
+          description: Successfully Created
+          schema:
+            $ref: '#/definitions/scope'
+        400:
+          description: Bad Request
+          schema:
+            $ref: '#/definitions/error'
+        409:
+          description: Conflict
+          schema:
+            $ref: '#/definitions/error'
+        500:
+          description: Server Error
+          schema:
+            $ref: '#/definitions/error'
+      tags:
+        - Scope Management
+
+    # Endpoint used to get all the available scopes
+    get:
+      description: |
+        This API is used to get all the available scopes.
+      x-wso2-request: |
+        curl -X GET -H "Authorization: Basic YWRtaW46YWRtaW4=" -H "Content-Type: application/json" "https://localhost:9443/api/identity/oauth2/v1.0/scopes"
+
+      x-wso2-response: |
+        [{"name":"profile","description":"get all profile information","displayName":"profile","bindings":["role1","role2"]}]
```

Comment:
Shouldn't we remove these wso2 specific swagger extensions going forward?

Reply:
+1 to remove